src/buffer/out/textBuffer.cpp

```diff
@@ -996,19 +996,29 @@ const COORD TextBuffer::GetWordStart(const COORD target, const std::wstring_view
     //  so the words in the example include ["word   ", "other  "]
     // NOTE: the start anchor (this one) is inclusive, whereas the end anchor (GetWordEnd) is exclusive
 
-    // can't expand left
-    if (target.X == GetSize().Left())
-    {
+#pragma warning(suppress : 26496)
+    // GH#7664: Treat EndExclusive as EndInclusive so
+    // that it actually points to a space in the buffer
+    auto copy{ target };
+    const auto bufferSize{ GetSize() };
+    if (target == bufferSize.Origin())
+    {
+        // can't expand left
         return target;
     }
+    else if (target == bufferSize.EndExclusive())
+    {
+        // treat EndExclusive as EndInclusive
+        copy = { bufferSize.RightInclusive(), bufferSize.BottomInclusive() };
+    }

```

Comment:
What if `target` is < `bufferSize.Origin()` or if `target` is > `bufferSize.EndExclusive()`?

Should we just be clamping this inside the bounds? Should we otherwise error out for those values?

Reply:
@miniksa I'm not sure -- we only have `Compare` in Viewport for assessing relative location of points and it `THROW`s if you try to compare something that's actually out of bounds. It seems unsafe to use it here given we're fixing an a11y crash

Reply:
Ick. Don't want that. If it's possible to do it with a `<=` and/or `>=` then fine. Otherwise, I withdraw my comment.

Reply:
Yeah. Gonna resolve this comment.
Side Note: Would it be possible/beneficial for us to create a non-crashing comparison operator for COORDs/points? It would look something like this:
```c++
bool operator<= (const COORD &c1, const COORD &c2)
{
    return c1.X <= c2.X && c1.Y <= c2.Y;
}
```
My main concern would be that it would cause some confusion with `CompareInBounds`, but this is just a thought.

Reply:
Maybe. But not right now. We already have a lot of ways of doing this comparison. If we're going to make new operators, I'd want to sit down and evaluate the whole codebase so we stop having like 12 utilities that do similar things.